lib/services/service_brokers/v2/client.rb

```diff
@@ -92,11 +92,15 @@ def fetch_service_instance_state(instance)
 
     def bind(binding)
       path = service_binding_resource_path(binding)
-      response = @http_client.put(path, {
-        service_id:  binding.service.broker_provided_id,
-        plan_id:     binding.service_plan.broker_provided_id,
-        app_guid:    binding.app_guid
-      })
+      attr = {
+          service_id:  binding.service.broker_provided_id,
+          plan_id:     binding.service_plan.broker_provided_id
+      }
+      if binding.respond_to? 'app_guid'
```

Comment:
There's a missing test case here. If you comment out the `if` and `end`, but leave the code that adds the app_guid to `attr`, you'll find that no tests fail. That means there is no test that drives out the desired change here: if the "binding" object does not respond to `#app_guid`, we want to exclude app_guid from the request.